AsyncDisplayKit/ASTableView.mm

```diff
@@ -903,6 +915,21 @@ - (void)rangeController:(ASRangeController *)rangeController didDeleteSectionsAt
     return; // if the asyncDataSource has become invalid while we are processing, ignore this request to avoid crashes
   }
 
+  if (_contentOffsetAdjustmentTopVisibleRow != nil) {
+    __block NSInteger adjustedSection = _contentOffsetAdjustmentTopVisibleRow.section;
+    __block NSInteger adjustedRow = _contentOffsetAdjustmentTopVisibleRow.row;
+    [indexSet enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL *stop) {
+        if (adjustedSection > idx) {
+          adjustedSection--;
+        } else if (adjustedSection == idx) {
+          adjustedRow = 0;
```

Comment:
From my understanding it's fine to have non-existent index in `_contentOffsetAdjustmentTopVisibleRow`, because it's never used to read nodes, it's used only to know the fact of being above or below first row.